src/fragments/lib/graphqlapi/js/working-with-files.mdx

```diff
@@ -0,0 +1,984 @@
+
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+async function addImageToSong(e: React.ChangeEvent<HTMLInputElement>) {
+  if (!currentSong) return;
+
+  // Upload the image to S3:
+  const key = await uploadImage(e);
+
+  const songDetails: UpdateSongInput = {
+    id: currentSong.id,
+    coverArtKey: key,
+  };
+
+  // Add the image to the current song:
+  try {
+    const updatedSong = await API.graphql<GraphQLQuery<UpdateSongMutation>>({
+      query: mutations.updateSong,
+      variables: { input: songDetails },
+    });
+    setCurrentSong(updatedSong?.data?.updateSong);
+
+    // Retrieve the image for the current song:
+    await getImageForCurrentSong();
+  } catch (error) {
+    console.error("Error adding image to song: ", error);
+  }
+}
+``` 
+
+</Block>
+<Block name="JavaScript">
+
+```js
+async function addImageToSong(e) {
+  if (!currentSong) return;
+
+  // Upload the image to S3:
+  const key = await uploadImage(e);
+
+  const songDetails = {
+    id: currentSong.id,
+    coverArtKey: key,
+  };
+
+  // Add the image to the current song:
+  try {
+    const updatedSong = await API.graphql({
+      query: mutations.updateSong,
+      variables: { input: songDetails },
+    });
+    setCurrentSong(updatedSong?.data?.updateSong);
+
+    // Retrieve the image for the current song:
+    await getImageForCurrentSong();
+  } catch (error) {
+    console.error("Error adding image to song: ", error);
+  }
+}
+```
+
+</Block>
+</BlockSwitcher>
+
+## Retrieving the associated file from the record
+
+To retrieve the file associated with a record, first query the record, then use Storage to get the signed URL. The signed URL can then be used to download the file, display an image, etc:
+
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+async function getImageForCurrentSong() {
+  // Query the record to get the file key:
+  const _song = await getSong();
+  // Check that the record has an associated image:
+  if (!_song?.coverArtKey) return;
+  // Retrieve the signed URL:
+  const signedURL = await Storage.get(_song?.coverArtKey);
+  setCurrentImageUrl(signedURL);
+}
+``` 
+
+</Block>
+<Block name="JavaScript">
+
+```js
+async function getImageForCurrentSong() {
+  // Query the record to get the file key:
+  const _song = await getSong();
+  // Check that the record has an associated image:
+  if (!_song?.coverArtKey) return;
+  // Retrieve the signed URL:
+  const signedURL = await Storage.get(_song?.coverArtKey);
+  setCurrentImageUrl(signedURL);
+}
+```
+
+</Block>
+</BlockSwitcher>
+
+## Deleting and removing associated files
+
+Once a file has been associated with a record, the file can be removed from the record, deleted from S3, or both.
```

Comment:
I think this warrants a bit more expansion. To me it reads like "there are three primary deletion workflows to when working with files and GraphQL API."
1. Remove both record and attached files
2. Keep the record but detach and delete the attached file
3. Keep the file but remove the record